onnx/checker.cc

```diff
@@ -228,6 +228,14 @@ void check_tensor(const TensorProto& tensor, const CheckerContext& ctx) {
 #undef check_field
 }
 
+void check_sequence(const SequenceProto& sequence) {
+  enforce_has_repeated_field(sequence, values);
```

Comment:
(a) enforce_has_repeated_fields requires that there be at least 1 repeated value. Do we want that condition for sequence/map here? That wouldn't allow an empty sequence to be an input, for example. May be better to remove that check?
(b) However, I think there are other sanity checks that could be done on the data: one check is to iterate through all elements of the sequence and calling the corresponding check function on each element; a second check is to verify that all elements are of the same type.

Reply:
Added the iterative checks, as well as helper functions for key value pair and sequence map element testing. Removed the has_repeated_field checks.